util/cache/cache.go

```diff
@@ -26,9 +26,6 @@ import (
 )
 
 const (
-	// ttl is the duration for which we keep entries in the cache.
-	ttl = 10 * time.Minute
```

Comment:
What about making this public as a DefaultTTL and use it instead of 10 * time.Minute all across the codebase?

Another option will be to have `New` and `NewWithTTL`

Reply:
I thought it would be good to force to specify the ttl everywhere given the high impact it has. I think ttl should be an explicit choice wherever we use the cache